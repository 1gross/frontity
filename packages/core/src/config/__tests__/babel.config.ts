import getBabel from '../babel.config';

test('Babel returns for development', () => {
  expect(getBabel('development')).toMatchSnapshot();
});

test('Babel returns for production', () => {
  expect(getBabel('production')).toMatchSnapshot();
});
