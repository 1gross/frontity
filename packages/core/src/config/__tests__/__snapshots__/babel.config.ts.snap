// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel returns for development 1`] = `
Object {
  "es5": Object {
    "plugins": Array [
      "transform-object-rest-spread",
      "transform-class-properties",
    ],
    "presets": Array [
      "typescript",
      Array [
        "env",
        Object {
          "targets": Object {
            "browsers": Array [
              "and_chr >= 67",
              "and_ff >= 18",
              "and_uc >= 11.8",
              "android >= 67",
              "not android <= 4.4.4",
              "chrome >= 49",
              "edge >= 12",
              "firefox >= 18",
              "ios_saf >= 10",
              "not op_mini all",
              "op_mob >= 46",
              "opera >= 36",
              "safari >= 10",
              "samsung >= 5",
            ],
          },
          "useBuiltIns": "usage",
        },
      ],
      "react",
    ],
  },
  "module": Object {
    "plugins": Array [
      "transform-object-rest-spread",
      "transform-class-properties",
      "react-hot-loader/babel",
    ],
    "presets": Array [
      "typescript",
      Array [
        "env",
        Object {
          "targets": Object {
            "esmodules": true,
          },
          "useBuiltIns": "usage",
        },
      ],
      "react",
    ],
  },
  "node": Object {
    "plugins": Array [
      "transform-object-rest-spread",
      "transform-class-properties",
    ],
    "presets": Array [
      "typescript",
      Array [
        "env",
        Object {
          "targets": Object {
            "node": "8.10",
          },
          "useBuiltIns": "usage",
        },
      ],
      "react",
    ],
  },
}
`;

exports[`Babel returns for production 1`] = `
Object {
  "es5": Object {
    "plugins": Array [
      "transform-object-rest-spread",
      "transform-class-properties",
    ],
    "presets": Array [
      "typescript",
      Array [
        "env",
        Object {
          "targets": Object {
            "browsers": Array [
              "and_chr >= 67",
              "and_ff >= 18",
              "and_uc >= 11.8",
              "android >= 67",
              "not android <= 4.4.4",
              "chrome >= 49",
              "edge >= 12",
              "firefox >= 18",
              "ios_saf >= 10",
              "not op_mini all",
              "op_mob >= 46",
              "opera >= 36",
              "safari >= 10",
              "samsung >= 5",
            ],
          },
          "useBuiltIns": "usage",
        },
      ],
      "react",
    ],
  },
  "module": Object {
    "plugins": Array [
      "transform-object-rest-spread",
      "transform-class-properties",
    ],
    "presets": Array [
      "typescript",
      Array [
        "env",
        Object {
          "targets": Object {
            "esmodules": true,
          },
          "useBuiltIns": "usage",
        },
      ],
      "react",
    ],
  },
  "node": Object {
    "plugins": Array [
      "transform-object-rest-spread",
      "transform-class-properties",
    ],
    "presets": Array [
      "typescript",
      Array [
        "env",
        Object {
          "targets": Object {
            "node": "8.10",
          },
          "useBuiltIns": "usage",
        },
      ],
      "react",
    ],
  },
}
`;
